### DISCUSSION TITLE
Cannot compile ImageMagick to use pthreads

### DISCUSSION THREAD

**Discussion by hcantunc** [ImageMagick/ImageMagick#6480] [General]
Hi, I'm trying to compile ImageMagick to use pthreads instead of OpenMP. I used the following command to compile but it did not work: 

`CFLAGS="-DMAGICKCORE_THREAD_SUPPORT=1" CXXFLAGS="-DMAGICKCORE_THREAD_SUPPORT=1"  ./configure --with-threads --disable-openmp`

I tried running certain commands such as `magick convert -limit thread 8 input output` (where the input is a large file) but the program does not display any concurrent behavior. Also, when I run `magick -list resource`, the number of available threads is displayed as 1. I also set the environment variable `MAGICK_THREAD_LIMIT` to 8 but that did not help. Is there a way to solve this issue?

Thanks.

**Comment by urban-warrior** [comment]
ImageMagick concurrency is managed with OpenMP.  Pthreads are only used for locks and synchronization.  For Pthreads, you'll need to create the threads in your favorite wrapper language, ImageMagick will make sure the threads are managed properly internally to ensure expected results.
